app/assets/javascripts/vendor/marker-generator.js

```diff
@@ -63,12 +59,12 @@ function MarkerGenerator(map, skateparks) {
     button.text(action.join(' '));
   }
 
-  function bindInfowindowListener(map, allMarkers, marker, infowindow) {
+  function bindInfowindowListener(marker) {
     marker.addListener('click', function() {
       allMarkers.forEach(function(marker){
         marker.infowindow.close();
       });
-      infowindow.open(map, marker);
+      marker.infowindow.open(map, marker);
```

Comment:
We should rename `bindInfoWindowListener` to something more explicit. How about `bindInfoWindowCloseOnMarkerClick` ? Still kinda jank but a little more detailed